### ISSUE TITLE
IDE0054 false positive in object initializer

### ISSUE THREAD

**Issue by Emdot** [dotnet/roslyn#52112]
**Version Used**: 
Microsoft Visual Studio Enterprise 2019
Version 16.9.2
.NET version 5.0.201

**Steps to Reproduce**:
```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<OutputType>Library</OutputType>
		<TargetFramework>net5.0</TargetFramework>
		<Nullable>enable</Nullable>
		<EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
	</PropertyGroup>
</Project>
```

```csharp
namespace N
{
    public class C
    {
        public string? I;
        public string J = "";
        public C Copy() => new() { I = I ?? J };
    }
}
```

**Expected Behavior**:
IDE0054 does not appear for `new() { I = I ?? J }`

**Actual Behavior**:
IDE0054 appears for `new() { I = I ?? J }`. Accepting the suggestion produces `new() { I ??= J }`, which is agrammatical.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/49303
